### ISSUE TITLE
Which location should we choose for MOOCs and other online events?

### ISSUE THREAD

**Issue by karindr** [hbz/oerworldmap#1261]
At the moment it is obligatory for events to give a location to them to save the entered data (we decided this because some stuff on the map works better with entries having locations).

Have you any suggestions which location we should choose for online events? The city of the provider, the country of the provider?

**Comment by formimkontext** [comment]
sounds reasonable.

**Comment by acka47** [comment]
We also had the problem with some organizations that are online communities and don't have a physical location. We might add an option "web" or "internet" as location. But then we will not be able to show them on the map and will have to think about how to show them instead...

**Comment by karindr** [comment]
A simmilar case: Open Education Week is kind of a collection of events online and all over the world. To be able to save it I chose USA but this is misleading. 

I think it would help in this cases if we enter a button for the second step like: "Save it anyway" and allow saving.

**Comment by trugwaldsaenger** [comment]
Good point. I saw the Open Education Week entry and found it quite irritating. We should try to find a solution for this. I could live with both proposals, adding a value "web" or allowing to save entries without location ("Save it anyway").

**Comment by karindr** [comment]
This is realy important because every time we have to do changes on such events without locations (for example on moocs) we are forced to enter a location to be able to safe the changes! We neeed a location "web" or a "save anyway" button.

**Comment by literarymachine** [comment]
> We neeed a location "web" or a "save anyway" button.

We not relax the schema and allow to simply save events without location? If this is not an option, the only approach that can quickly be implemented is the `Web` location, see https://github.com/hbz/oerworldmap/issues/1237#issuecomment-334085048.

**Comment by karindr** [comment]
It would be very helpful to have a desicion on this.

**Comment by trugwaldsaenger** [comment]
> Have you any suggestions which location we should choose for online events? The city of the provider, the country of the provider?

I guess the address of the provider would be the best in many cases. If, for example Delft University provides some mooc, the mooc should have the address of Delft University...

> Why not relax the schema and allow to simply save events without location?

Would be OK for me as well.

**Comment by karindr** [comment]
If we establish web as location we give the possibility to search for events with this location, so the visitors can find all online events. If we simply skip the need to enter a location there is no distinction between online events and events the editor forgot to add a location. So I suggest to add web what can be quickly done as @literarymachine writes above.

**Comment by literarymachine** [comment]
> So I suggest to add web what can be quickly done as @literarymachine writes above.

The only thing I see here is that if we add `Web` as a location, should this not be added to all / most `Service`s, too? We added the whole "inheritance" of locations just to deal with things that don't have a location of their own...

**Comment by trugwaldsaenger** [comment]
> The only thing I see here is that if we add Web as a location, should this not be added to all / most Services, too? We added the whole "inheritance" of locations just to deal with things that don't have a location of their own

Sounds reasonable to me. But what would be the consequences? I think it is helpful, that a service, which is provided by hbz can be counted for the land of Northrhine Westphalia. Also I think that services and events should be shown in the popup menue, when hovering over the pin of the institution. Would all this be affected?

**Comment by literarymachine** [comment]
> Please don't forget, that events are the exeption which are not inheriting the location from their organisers!

Following what @trugwaldsaenger says in https://github.com/hbz/oerworldmap/issues/1261#issuecomment-398314959, I think this exception is wrong. Events should be counted in country statistics, and this would work by inheriting the location.

**Comment by literarymachine** [comment]
- [ ] Schema: make `location` optional
- [ ] Checkbox to indicate if `online` or `offline`
